app/front_end/OTTOfficeTimeTracker/app/src/main/java/com/example/officetimetracker/RegisterActivity.java

```diff
@@ -3,15 +3,53 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 import android.content.Intent;
+import android.content.SharedPreferences;
 import android.os.Bundle;
 import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
 
 public class RegisterActivity extends AppCompatActivity {
 
+    EditText email, username,pass;
+    Button register;
+
+    SharedPreferences preferences;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_register);
+        email=findViewById(R.id.emailReg);
+        username=findViewById(R.id.nameReg);
+        pass=findViewById(R.id.passReg);
+        register=findViewById(R.id.btnSubmitReg);
+
+        preferences=getSharedPreferences("UserInfo",0);
+
+        register.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View view){
+                String emailValue=email.getText().toString();
+                String userValue=username.getText().toString();
+                String passValue=pass.getText().toString();
+
+                if(userValue.length()>1 && emailValue.length()>1 && passValue.length()>1) {
+                    SharedPreferences.Editor editor = preferences.edit();
```

Comment:
I suggest to validate the e-mail first, to have the pattern nnnn@ppp.ddd
Then base on email (which should be already unique) search in storage if the user is already registered.
only now you can proceed with the register.